src/main/scala/io/iohk/ethereum/blockchain/sync/PeersClient.scala

```diff
@@ -0,0 +1,151 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Cancellable, Props, Scheduler}
+import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+import io.iohk.ethereum.network.{Peer, PeerId}
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.network.p2p.messages.PV63.{GetNodeData, NodeData}
+import io.iohk.ethereum.network.p2p.{Message, MessageSerializable}
+import io.iohk.ethereum.utils.Config.SyncConfig
+
+import scala.concurrent.ExecutionContext
+import scala.reflect.ClassTag
+
+class PeersClient(
+    val etcPeerManager: ActorRef,
+    val peerEventBus: ActorRef,
+    val syncConfig: SyncConfig,
+    implicit val scheduler: Scheduler
+) extends Actor
+    with ActorLogging
+    with PeerListSupport
+    with BlacklistSupport {
+  import PeersClient._
+
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  val statusSchedule: Cancellable =
+    scheduler.schedule(syncConfig.printStatusInterval, syncConfig.printStatusInterval, self, PrintStatus)
+
+  def receive: Receive = running(Map())
+
+  override def postStop(): Unit = {
+    super.postStop()
+    statusSchedule.cancel()
+  }
+
+  def running(requesters: Map[ActorRef, ActorRef]): Receive =
+    handleBlacklistMessages orElse handlePeerListMessages orElse {
+      case PrintStatus =>
+        log.debug(
+          """PeersClient status:
+          | requests: {}
+          | available peers: {}""".stripMargin,
+          requesters.size,
+          peersToDownloadFrom.size)
+      case BlacklistPeer(peerId, reason) => peerById(peerId).foreach(blacklistIfHandshaked(_, reason))
+      case Request(message, peerSelector, toSerializable) =>
+        val requester = sender()
+        selectPeer(peerSelector) match {
+          case Some(peer) =>
+            val handler =
+              makeRequest(peer, message, responseMsgCode(message), toSerializable)(scheduler, responseClassTag(message))
+            val newRequesters = requesters + (handler -> requester)
+            context become running(newRequesters)
+          case None =>
+            log.debug("No suitable peer found to issue a request")
+            requester ! NoSuitablePeer
+        }
+      case PeerRequestHandler.ResponseReceived(peer, message, _) =>
+        val requestHandler = sender()
```

Comment:
code under `PeerRequestHandler.ResponseReceived` is almost duplication of `PeerRequestHandler.RequestFailed(peer, reason)`, maybe extract it to separate method ?

Reply:
Fixed